### PR TITLE
feat(visitors-config): Enable media on promotion.

### PR DESCRIPTION
**Pull request by hristoterezov**

### CODE REVIEW

**Review by damencho — DISMISSED** [review]

File: react/features/base/conference/actions.ts [hunk 1]
```diff
@@ -1059,13 +1061,31 @@ export function redirect(vnode: string, focusJid: string, username: string) {
...
                 dispatch(gumPending([ MEDIA_TYPE.AUDIO, MEDIA_TYPE.VIDEO ], IGUMPendingState.NONE));
 
                 // FIXME: Workaround for the web version. To be removed once we get rid of conference.js
                 if (typeof APP !== 'undefined') {
+                    if (!vnode) {
+                        const { enableMediaOnPromote = {} } = getState()['features/base/config'].visitors ?? {};
+                        const { audio = false, video = false } = enableMediaOnPromote;
+
+                        if (audio) {
+                            dispatch(setAudioMuted(false, true));
```
Comment (jallamsetty1) [inline react/features/base/conference/actions.ts:1075]: What if the visitor initially joined with startWithAudioMuted/startWithVideoMuted or selected audio-only mode on join? Shouldn't we take those into consideration before unmuting them?

Comment (hristoterezov) [reply react/features/base/conference/actions.ts:1075]: hmm... We discussed we will ignore those for now. but I agree this is kind of an edge case. lets discuss offline with the team.

Comment (hristoterezov) [reply react/features/base/conference/actions.ts:1075]: actually I think audio only should work and unmute, I have to test it. But muting from the prejoin would be ignored. Otherwise if the config option is set we already discussed that the new properties will be higher priority.

Comment (hristoterezov) [reply react/features/base/conference/actions.ts:1075]: Will fix it

Comment (hristoterezov) [reply react/features/base/conference/actions.ts]: Fixed!

**Review by damencho — APPROVED** [review]

**Review by jallamsetty1 — APPROVED** [review]
